e2e/e2e_test.go

```diff
@@ -81,23 +97,253 @@ var _ = Describe("Whereabouts functionality", func() {
+	if err != nil {
+		return err
+	}
+
+	ipPoolAllocations := ipPool.Allocations()
+	podList, err := ListPods(cs, testNamespace, wbLabelEqual)
+	if err != nil {
+		return err
+	}
+
+	if err := foundPodIPforAllIPPoolIPs(ipPoolAllocations, podList); err != nil {
+		return err
+	}
+
+	if err := foundIPPoolIPforAllPodIPs(ipPoolAllocations, podList); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func foundPodIPforAllIPPoolIPs(ipPoolAllocations []types.IPReservation, podList *core.PodList) error {
+	var found bool
+	var podIP, reservedIP string
+	var err error
+
+	for _, allocation := range ipPoolAllocations {
+		found = false
+		for _, pod := range podList.Items {
+			podIP, err = secondaryIfaceIPValue(&pod)
+			if err != nil {
+				return err
+			}
+			reservedIP = allocation.IP.String()
+
+			if reservedIP == podIP {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			return fmt.Errorf("Possible stale IP Pool: failed to find pod for IP Pool IP %s\n", allocation.IP)
+		}
+	}
+
+	return nil
+}
+
+func foundIPPoolIPforAllPodIPs(ipPoolAllocations []types.IPReservation, podList *core.PodList) error {
+	var found bool
+	var podIP, reservedIP string
+	var err error
+
+	for _, pod := range podList.Items {
+		found = false
+		for _, allocation := range ipPoolAllocations {
+			podIP, err = secondaryIfaceIPValue(&pod)
+			if err != nil {
+				return err
+			}
+			reservedIP = allocation.IP.String()
+
+			if reservedIP == podIP {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			return fmt.Errorf("possible pod IP not recorded in IP pool: failed to find IP pool allocation for pod IP %s\n", podIP)
+		}
+	}
+
+	return nil
+}
+
+type envVars struct {
+	kubeconfigPath      string
+	numComputeNodes     int
+	fillPercentCapacity int
+	numThrashIter       int
+}
+
+func environment() (envVars, error) {
 	const kubeconfig = "KUBECONFIG"
+	const numCompute = "NUMBER_OF_COMPUTE_NODES"
+	const fillCapcity = "FILL_PERCENT_CAPACITY"
+	const numThrash = "NUMBER_OF_THRASH_ITER"
+
+	var err error
+
+	envVars := envVars{}
 
 	kubeconfigPath, found := os.LookupEnv(kubeconfig)
 	if !found {
-		return nil, fmt.Errorf("must provide the path to the kubeconfig via the `KUBECONFIG` env variable")
+		kubeconfigPath = "${HOME}/.kube/config"
+	}
+	numComputeNodes, found := os.LookupEnv(numCompute)
+	if !found {
+		numComputeNodes = "2"
+	}
+	fillPercentCapacity, found := os.LookupEnv(fillCapcity)
+	if !found {
+		fillPercentCapacity = "50"
+	}
+	numThrashIter, found := os.LookupEnv(numThrash)
+	if !found {
+		numThrashIter = "1"
+	}
+
+	envVars.kubeconfigPath = kubeconfigPath
+	envVars.numComputeNodes, err = strconv.Atoi(numComputeNodes)
+	if err != nil {
+		return envVars, err
+	}
+	envVars.fillPercentCapacity, err = strconv.Atoi(fillPercentCapacity)
+	if err != nil {
+		return envVars, err
+	}
+	envVars.numThrashIter, err = strconv.Atoi(numThrashIter)
+	if err != nil {
+		return envVars, err
 	}
```

Comment:
I would actually create an helper each for compute nodes, percent capacity, and num iterations, then use it.

For instance:
```golang
func environment() (envVars, error) {
    ...
    numComputeNodes, err := computeNodes()
    if err != nil {
        return envVars{}, err
    }
    ...
    return envVars{
        numComputeNodes: numComputeNodes,
        ...
    }
}

func computeNodes() (int, error) {
        numComputeNodes, found := os.LookupEnv(numCompute)
	if !found {
		numComputeNodes = "2"
	}
        return strconv.Atoi(numComputeNodes)
}
```

Furthermore, `return envVars, err` is not as robust as it could be - imagine the caller forgets to check the error. He would have an half created env vars, which surely would be hard to debug. 

You can make the code more resilient for returning `nil, err` - you'd have to adjust the return type, making it a pointer. Or you could alternatively return an empty envVars struct.